modules/api/api-xsjs/src/main/java/com/sap/xsk/api/destination/CloudPlatformDestinationFacade.java

```diff
@@ -0,0 +1,126 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.api.destination;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.sap.cloud.sdk.cloudplatform.CloudPlatformAccessor;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.cloudplatform.connectivity.DestinationAccessor;
+import com.sap.cloud.sdk.cloudplatform.connectivity.HttpClientAccessor;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.MethodNotSupportedException;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.client.methods.HttpOptions;
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.methods.HttpTrace;
+import org.apache.http.util.EntityUtils;
+import org.eclipse.dirigible.commons.api.scripting.IScriptingFacade;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+
+public class CloudPlatformDestinationFacade implements IScriptingFacade {
+
+  private static String QUERY_PATH = "queryPath";
+  private static String METHOD = "method";
+  private static String HEADERS = "headers";
+  private static String STATUS_CODE = "statusCode";
+  private static String TEXT = "text";
+  private static String NAME = "name";
+  private static String VALUE = "value";
+
+  public static Destination getDestination(String name) {
+    setKymaCloudPlatformFacade();
+
+    return DestinationAccessor.getDestination(name);
+  }
+
+  public static String executeRequest(String requestObject, Destination destination) throws IOException, MethodNotSupportedException {
+    setKymaCloudPlatformFacade();
+
+    JsonParser jsonParser = new JsonParser();
+    JsonObject requestJson = jsonParser.parse(requestObject).getAsJsonObject();
+
+    HttpClient client = HttpClientAccessor.getHttpClient(destination.asHttp());
+    HttpRequestBase request = getRequest(requestJson);
+
+    request.setURI(URI.create(destination.asHttp().getUri() + requestJson.get(QUERY_PATH).getAsString()));
+    setRequestHeaders(requestJson, request);
+
+    HttpResponse response = client.execute(request);
+
+    JsonObject responseJson = new JsonObject();
+    JsonArray headers = new JsonArray();
+    Arrays.stream(response.getAllHeaders()).map(CloudPlatformDestinationFacade::createHeaderJson).forEach(headers::add);
+
+    responseJson.add(HEADERS, headers);
+    responseJson.addProperty(STATUS_CODE, response.getStatusLine().getStatusCode());
+    responseJson.add(TEXT, jsonParser.parse(EntityUtils.toString(response.getEntity())));
```

Comment:
this fails if response is not JSON. Either don't parse it, or setLenient (not preferred)